library/src/main/java/com/datatorrent/lib/db/cache/CacheManager.java

```diff
@@ -210,6 +272,54 @@ public String getRefreshTime()
     Map<Object, Object> loadInitialData();
   }
 
+
+  /**
+   * Used by {@link CacheManager} to pass properties to its stores which implement {@link Component}.
+   */
+  public static class CacheContext implements Context
+  {
+    public static final transient Attribute<Boolean> READ_ONLY_ATTR = new Attribute<>((false));
+    public static final transient Attribute<Integer> NUM_INIT_CACHED_LINES_ATTR = new Attribute<>((-1));
```

Comment:
The checks in other places might be straightforward if it is initialized to Integer.MAX_VALUE.